wasm/src/bits-builtins.c

```diff
@@ -0,0 +1,214 @@
+#include "arithmetic.h"
+#include "bits.h"
+#include "mpd.h"
+
+#define swap(x, y) { mpd_t *v = x; x = y; y = v; }
+
+opa_value *opa_bits_or(opa_value *a, opa_value *b)
+{
+    mpd_t *x = opa_number_to_bf(a);
+    mpd_t *y = opa_number_to_bf(b);
+    if (x == NULL || y == NULL)
+    {
+        opa_mpd_del(x);
+        opa_mpd_del(y);
+        return NULL;
+    }
+
+    if (mpd_sign(x) == mpd_sign(y))
+    {
+        if (mpd_sign(x)) // x neg
+        {
+            // (-x) | (-y) == ^(x-1) | ^(y-1) == ^((x-1) & (y-1)) == -(((x-1) & (y-1)) + 1)
+            mpd_t *x1 = qsub_one(qabs(x));
+            mpd_t *y1 = qsub_one(qabs(y));
+            mpd_t *z = qadd_one(qand(x1, y1));
+            return opa_bf_to_number(qneg(z));
+        }
+
+        // x | y == x | y
+        return opa_bf_to_number(qor(x, y));
+    }
+
+    // x.neg != y.neg
+	if (mpd_sign(x))
+    {
```

Comment:
nit: Looks like mixed tabs/spaces

💭 maybe we should look into a linter/style check on the wasm c files as they are growing quite a bit

Reply:
Fixed.